src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs

```diff
@@ -46,11 +47,18 @@ public ConnectionFactory(ConnectionConfiguration connectionConfiguration)
             }
         }
 
+        public IConnection CreatePublishConnection() => CreateConnection("Publish");
```

Comment:
The caching is now happening in the `ChannelProvider` so there's no need for a separate class to do this.